src/dist/replication/meta_server/duplication/duplication_info.cpp

```diff
@@ -191,5 +191,25 @@ duplication_info_s_ptr duplication_info::decode_from_blob(dupid_t dup_id,
     return dup;
 }
 
+void duplication_info::append_if_valid_for_query(
+    const app_state &app,
+    /*out*/ std::vector<duplication_entry> &entry_list) const
+{
+    zauto_read_lock l(_lock);
+
+    entry_list.emplace_back(to_duplication_entry());
+    duplication_entry &ent = entry_list.back();
+    ent.__isset.pending = true;
+    // the confirmed decree is not useful for displaying
+    // the overall state of duplication, instead we show pending mutations.
+    ent.__isset.progress = false;
+    for (const partition_configuration &part : app.partitions) {
+        int pid = part.pid.get_partition_index();
+        auto it = _progress.find(pid);
+        int64_t pending = part.last_committed_decree - it->second.stored_decree;
+        ent.pending[pid] = std::max(pending, int64_t(0));
```

Comment:
Why the variable of ent.progress is not set here?

Reply:
Because progress (confirmed_decree) does not give useful information in terms of querying duplication state. We need "pending" to tell whether the duplication is healthy or blocked due to duplication failure, eg. intra-cluster network failure.